### ISSUE TITLE
Folders created with special characters get renamed

### ISSUE THREAD

**Issue by HarshaVardhanJ** [Uzay-G/archivy#53]
### Sequence of steps:

* Created a new folder with the name `Folder%$'` from the UI.
* Tried to delete the folder from the UI(Unable to access the delete icon because the `Expand` link does not work).
* Unable to delete the folder.
* Refreshed the page.
* New folder now shows up as `Folder`(is renamed by Archivy).
* The folder can now be deleted from the UI.

Nothing relevant shows up in the logs either. In any case, I've attached the logs below:
```
Setting environment variables.
The following environment variables were set:
		FLASK_DEBUG=1
		ELASTICSEARCH_ENABLED=0
		ELASTICSEARCH_URL=http://localhost:9200/
Elasticsearch not used. Search function will not work.
Starting Archivy
 * Serving Flask app "archivy" (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: on
 * Running on http://0.0.0.0:5000/ (Press CTRL+C to quit)
 * Restarting with inotify reloader
 * Debugger is active!
 * Debugger PIN: 329-306-783
172.17.0.1 - - [26/Aug/2020 06:49:01] "GET /index HTTP/1.1" 200 -
172.17.0.1 - - [26/Aug/2020 06:49:08] "POST /folders/new HTTP/1.1" 200 -
172.17.0.1 - - [26/Aug/2020 06:50:06] "GET /index HTTP/1.1" 200 -
172.17.0.1 - - [26/Aug/2020 06:50:10] "DELETE /folders/delete HTTP/1.1" 200 -
```

I'm guessing the folder names are being sanitised before being added and certain special characters are being removed. When I check the directory in which Archivy saves its files, the new folder(`Folder%$'`) is being stored as `Folder`.

**Comment by Uzay-G** [comment]
Do you think we should not sanitize the special characters? That is one issue, but there is also the problem where deleting is not working. Ill check that out.

**Comment by srevinsaju** [comment]
I guess we should make the folder names urlsafe. Because a user could add a `/` which would make python think that its a directory

````python
def slugify(value):
    """
    Normalizes string, converts to lowercase, removes non-alpha characters,
    and converts spaces to hyphens.
    """
    import unicodedata
    value = unicodedata.normalize('NFKD', value).encode('ascii', 'ignore')
    value = unicode(re.sub('[^\w\s-]', '', value).strip().lower())
    value = unicode(re.sub('[-\s]+', '-', value))
    # ...
    return value
```
This is from the Django source code as of https://stackoverflow.com/questions/295135/turn-a-string-into-a-valid-filename

The answer recommends as the gold standard of escaping file names;

**Comment by Uzay-G** [comment]
We're currently using this:

```python
def valid_filename(name):
    name = str(name).strip().replace(" ", "_")
    return re.sub(r"(?u)[^-\w.]", "", name)
```

**Comment by Uzay-G** [comment]
That one could also work though

**Comment by clemux** [comment]
werkzeug, used by flask, includes [werkzeug.utils.secure_filename](https://werkzeug.palletsprojects.com/en/1.0.x/utils/#werkzeug.utils.secure_filename)

**Comment by HarshaVardhanJ** [comment]
@Uzay-G Supposing somebody wants to create two folders, both of which contain special characters as shown in the original post, albeit in a different arrangement, will it not cause a conflict when they're saved on the host?

For example,

`Folder%$'` and `Folder$%' ` would both be saved under the name `Folder` on the host.

**Comment by HarshaVardhanJ** [comment]
@Uzay-G Just did the following:
- Created a folder with the name `Test%$'`
- Created another folder with the name `Test$%'`.
- Checked the name of the folders saved in the `data` directory. Only one folder by the name `Test` is present.
- Upon refreshing the page, I can see just one folder entry titled `Folder`.

**Comment by Uzay-G** [comment]
https://github.com/Uzay-G/archivy/commit/64dc1eaa71c060fbf1e090f11158f72e8804ff2e should introduce some fixes to these issues. Could you check if you encounter the same issues?

**Comment by HarshaVardhanJ** [comment]
I think the issues have been fixed. Here's a list of actions taken with the results:

1. Created a folder titled `Folder$%'`.
2. It was saved as `Folder`. The name reflected immediately in the UI. There was no necessity to refresh the page unlike last time.
3. I tried to create another folder titled `Folder%$'`. This wasn't accepted and the message `Directory already exists` was displayed in the text box.
3. I created another folder titled `Folder$55` inside the first folder(`Folder`). It was saved as `Folder55` inside the first folder.
4. Upon checking the `data` directory, the folder names were as it was displayed in the UI(`data/Folder/Folder55`).

Also, I was able to expand the folder and delete them right after creating them. No issues there!

I have one suggestion. Would it be possible to try not to get rid of the special characters in the folder/file names?
